llvm/lib/Target/AArch64/GISel/AArch64InstructionSelector.cpp

```diff
@@ -7985,6 +7996,29 @@ bool AArch64InstructionSelector::isLoadStoreOfNumBytes(
   return (*MI.memoperands_begin())->getSize() == NumBytes;
 }
 
+bool AArch64InstructionSelector::isTop16Zero(const MachineOperand &MO) const {
```

Comment:
The interface is the MachineOperand, not the register. So maybe rename this from "RegPredicate"? LeafPredicate?